includes/class-wc-subscriptions-order.php

```diff
@@ -622,26 +622,35 @@ public static function get_users_subscription_orders( $user_id = 0 ) {
 		}
 
 		// Get all the customers orders which are not subscription renewal orders
-		$order_ids = get_posts( array(
-			'posts_per_page' => 1,
-			'post_type'      => 'shop_order',
-			'post_status'    => 'any',
-			'fields'         => 'ids',
-			'orderby'        => 'date',
-			'order'          => 'DESC',
-			'meta_query'     => array(
-				array(
-					'key'     => '_customer_user',
-					'compare' => '=',
-					'value'   => $user_id,
-					'type'    => 'numeric',
-				),
-				array(
+		$custom_query_var_handler = function( $query, $query_vars ) {
+			if ( ! empty( $query_vars['_non_subscription_renewal'] ) ) {
+				$query['meta_query'][] = array(
 					'key'     => '_subscription_renewal',
 					'compare' => 'NOT EXISTS',
-				),
-			),
-		) );
+				);
+				unset( $query_vars['_non_subscription_renewal'] );
+			}
+
+			return $query;
+		};
+		add_filter( 'woocommerce_order_data_store_cpt_get_orders_query', $custom_query_var_handler, 10, 2 );
+
+		$all_possible_statuses = array_values( array_unique( array_keys( wc_get_order_statuses() ) ) );
```

Comment:
Shouldn't it be:
```suggestion
		$all_possible_statuses = array_unique( array_keys( wc_get_order_statuses() ) );
```
To me `array_values` seems redundant here...

Reply:
Sorry I missed your comment before merging this PR. Thank you for suggestion.

However, I think `array_values()` is necessary to avoid non-incremental array index.

Consider this:
```php
<?php
$a = ['a', 'b', 'c', 'a', 'd'];
var_dump(array_unique($a));
var_dump(array_values(array_unique($a)));
```

Output is:
```
array(4) {
  [0]=>
  string(1) "a"
  [1]=>
  string(1) "b"
  [2]=>
  string(1) "c"
  [4]=>
  string(1) "d"
}
array(4) {
  [0]=>
  string(1) "a"
  [1]=>
  string(1) "b"
  [2]=>
  string(1) "c"
  [3]=>
  string(1) "d"
}
```

Notice the first output array is has no array index 3.

Reply:
Why do we care about indexes? keys / indexes are irrelevant in this context.

Ref: https://github.com/woocommerce/woocommerce/blob/6.5.0/plugins/woocommerce/includes/data-stores/class-wc-order-data-store-cpt.php#L782

Reply:
After some exploration, I think even `array_unique` is irrelevant...

```
		$all_possible_statuses = array_values( array_unique( array_keys( wc_get_order_statuses() ) ) );

		$args = array(
			'type'        => 'shop_order',
			'status'      => $all_possible_statuses,
			'orderby'     => 'date',
			'order'       => 'DESC',
			'customer_id' => $user_id,
			'return'      => 'ids',
		);
```

could have been
```
		$args = array(
			'type'        => 'shop_order',
			'status'      => array_keys( wc_get_order_statuses() ),
			'orderby'     => 'date',
			'order'       => 'DESC',
			'customer_id' => $user_id,
			'return'      => 'ids',
		);
```

Reply:
You're right that it might not matter but not calling `array_unique()` or `array_values()` isn't much of an optimization. I won't know how that values would be used down the line now or even more so later, so I thought just to be safe, I'd pass a proper array with proper index and non-duplicate values.

Reply:
> You're right that it might not matter but not calling `array_unique()` or `array_values()` isn't much of an optimization.

It can have impact on performance, but for me it was more off cognitive load, than some optimization concern.